### DISCUSSION TITLE
Is there a good way to update the global "lang" attribute?

### DISCUSSION THREAD

**Discussion by Natr1x** [sveltejs/kit#8757] [Q&A] [answered]
The accessibility section of the docs recommends using server hooks with `transformPageChunk` in  order to support multiple languages but that only works if you are actually making server requests. Is there any way to do it that works for both server and client side? 

Something like `<svelte:head />` maybe?

**Comment by theetrain** [comment, marked as answer]
I put together this demo that makes use of form actions to change the page's language: https://www.sveltelab.dev/xya98neroqbjl4t

As long as you have server side rendering, this solutions works with and without JS.

<details>
<summary>Example</summary>

app.html
```html
<html lang="%lang%">
```

+page.svelte
```html
<form method="POST" use:enhance>
	<button formaction="?l=en">English</button>
	<button formaction="?l=zh">中文</button>
</form>
```

+page.server.js
```js
import { redirect } from '@sveltejs/kit'

export const load = async ({ cookies, depends }) => {
	depends('site:lang')
	const lang = cookies.get('lang')
	console.log('loading lang', {lang})

	let title = 'Language switcher!'

	if (lang === 'zh') {
		title = '语言切换器'
	}
	return {title};
};

const allowedLanguages = ['en', 'zh']

/** @type {import('./$types').Actions} */
export const actions = {
	default: async ({ url, cookies }) => {
		const lang = url.searchParams.get('l')

		console.log({lang})
		if (allowedLanguages.includes(lang)) {
			cookies.set('lang', lang, { path: '/' })
		}

		// redirect is necessary for full page load
		throw redirect(303, '/')
	}
};
```

hooks.server.js
```js
/** @type {import('./$types').Handle */
export async function handle({ event, resolve }) {
	const lang = event.cookies.get('lang') || 'en'
	console.log('hook lang is', lang)
	const response = await resolve(event, {
		transformPageChunk: ({html}) => html.replace('%lang%', lang)
	});
	return response;
}
```

</details>

For a pure client-side solution, you can set the document language only using JS with:

```js
document.documentElement.lang = 'language-code'
```

**Reply by Natr1x** [reply]
I don't remember if this would have solved my original problem but I think so. I like the use of a form, very elegant.
